### PR TITLE
[14.0][IMP] cetmix_tower_server: Server Template mods

### PR DESCRIPTION
**Pull request by ivs-cetmix**
Improve UX by adding new features and modifying existing ones.

Server template and wizard
--------------------------

- Variable values can be edited in wizard before server creation
- Make some template fields not required because they can be set later
- Add 'Create Server' action in the kanban tile 'Action' menu
- Improve server template wizard layout

Server
------

Add context key that allows to skip constraint checks for ssh fields. This is needed in order to avoid exceptions when creating a new server from a server template using wizard.

While it's possible to modify server properties from a flight plan that is called after server is created those SSH fields can be populated later. So there is no need to block them.

Task: 4027
Doc: 2

## Release Notes

- **New Features**
	- Enhanced SSH settings validation with a new context key for optional checks.
	- Simplified `ssh_port` field in server templates, making it non-mandatory.
	- Added support for additional configuration variables during server creation.
	- Introduced a color field for better visualization in the server template wizard.

- **User Interface Improvements**
	- Updated kanban view with a "Create Server" action link for better management.
	- Reorganized form layout with new fields for improved user interaction, including a color picker and editable variable entries.
	- Updated `ssh_username` field with a placeholder indicating it can be changed later.

- **Bug Fixes**
	- Improved error handling for SSH settings validation, allowing multiple issues to be reported at once.

### CODE REVIEW

File: cetmix_tower_server/models/cx_tower_server.py [hunk 1]
```diff
@@ -403,20 +403,27 @@ def _compute_file_count(self):
...
-                )
-            if rec.ssh_auth_mode == "k" and not rec.ssh_key_id:
-                raise ValidationError(
-                    _("Please provide SSH Key for %(srv)s", srv=rec.name)
-                )
+        """Ensure SSH settings are valid.
+        Set 'skip_ssh_settings_check' context key to skip the checks
+        """
+
+        if not self._context.get("skip_ssh_settings_check"):
```
Comment (GabbasovDinar) [inline cetmix_tower_server/models/cx_tower_server.py]: in order to reduce code nesting you can make a positive check:

```python
if self._context.get("skip_ssh_settings_check"):
    return False

for rec in self:
    ...
```

**Review by GabbasovDinar — CHANGES_REQUESTED** [review]

File: cetmix_tower_server/models/cx_tower_server.py (hunk 1 above)
Comment (ivs-cetmix) [reply cetmix_tower_server/models/cx_tower_server.py]: @GabbasovDinar I thought about it however this is a constrain function that doesn't return anything. So I decided to go with a negative check.

**Review by GabbasovDinar — APPROVED** [review]

**Review by Aldeigja — APPROVED** [review]
Functional LGTM

**Comment by CetmixGitDrone** [comment]
Hey, thanks for contributing! Proceeding to merge this for you.
Prepared branch [14.0-dev-ocabot-merge-pr-133-by-ivs-cetmix-bump-patch](https://github.com/cetmix/cetmix-tower/commits/14.0-dev-ocabot-merge-pr-133-by-ivs-cetmix-bump-patch), awaiting test results.

**Comment by CetmixGitDrone** [comment]
Congratulations, your PR was merged at 55174055b253b63de86d9029d15a2828174eadbe. Thanks a lot for contributing to cetmix. ❤️

**Pull request merged**